core/src/cubos/core/data/ser/json.cpp

```diff
@@ -0,0 +1,97 @@
+#include <cubos/core/log.hpp>
+
+/// [JSON serializer]
+#include <cubos/core/data/ser/json.hpp>
+
+using cubos::core::data::JSONSerializer;
+using cubos::core::reflection::Type;
+
+auto jsonVal = nlohmann::json::object(); ///< The current JSON value that is being serialized.
```

Comment:
Do not use a global variable. Instead, write directly to `this->mJSON` on the hooks.